### PR TITLE
Extend explanations in quantize doc and add 32-bit baseline

### PR DESCRIPTION
**Pull request by rasbt**
I added a 32-bit baseline and extended the explanations a bit.

### CODE REVIEW

File: tutorials/quantize.md [hunk 1]
```diff
@@ -1,14 +1,32 @@
...
+
+
+```bash
+python generate/base.py --checkpoint_dir checkpoints/tiiuae/falcon-7b --precision 32-full --max_new_tokens 256
+...
+Time for inference 1: 10.69 sec total, 23.95 tokens/sec.
+Memory used: 28.95 GB
+```bash
+
+First, using a lower precision compared to 32-bit float can result in two times reduced memory consumption. You can either try setting `--precision 16-full` for regular 16-bit precision or  `--precision bf16-full` if your GPU supports brain-float 16-bit precision. ([This brief video](https://lightning.ai/courses/deep-learning-fundamentals/9.0-overview-techniques-for-speeding-up-model-training/unit-9.1-accelerated-model-training-via-mixed-precision-training/) explains the difference between regular 16-bit and bf16-bit precision.)
```
Comment (carmocca) [inline tutorials/quantize.md:23]:
```suggestion
First, using a lower precision compared to 32-bit float can result in two times reduced memory consumption. You can either try setting `--precision 16-true` for regular 16-bit precision or  `--precision bf16-true` if your GPU supports brain-float 16-bit precision. ([This brief video](https://lightning.ai/courses/deep-learning-fundamentals/9.0-overview-techniques-for-speeding-up-model-training/unit-9.1-accelerated-model-training-via-mixed-precision-training/) explains the difference between regular 16-bit and bf16-bit precision.) 
```

File: tutorials/quantize.md [hunk 2]
```diff
@@ -1,14 +1,32 @@
...
 
+
 ### Baseline
 
-All the examples below were run on an A100 40GB GPU.
+It's useful to start with a baseline to have a reference point for memory savings via the various quantization methods.
+
+
+```bash
+python generate/base.py --checkpoint_dir checkpoints/tiiuae/falcon-7b --precision 32-full --max_new_tokens 256
```
Comment (carmocca) [inline tutorials/quantize.md:17]:
```suggestion
python generate/base.py --checkpoint_dir checkpoints/tiiuae/falcon-7b --precision 32-true --max_new_tokens 256
```

**Review by carmocca — APPROVED** [review]
Thank you!

File: tutorials/quantize.md [hunk 3]
```diff
@@ -1,14 +1,32 @@
...
-All the examples below were run on an A100 40GB GPU.
+It's useful to start with a baseline to have a reference point for memory savings via the various quantization methods.
+
+
+```bash
+python generate/base.py --checkpoint_dir checkpoints/tiiuae/falcon-7b --precision 32-true --max_new_tokens 256
+...
+Time for inference 1: 10.69 sec total, 23.95 tokens/sec.
+Memory used: 28.95 GB
+```bash
```
Comment (Andrei-Aksionov) [inline tutorials/quantize.md:21]: It looks like "```bash" signals to start a new code block which results in a weird-looking code/text mixture.

Screenshot 2023-07-26 at 2 16 03 PM

File: tutorials/quantize.md [hunk 4]
```diff
@@ -1,14 +1,32 @@
...
+
+First, using a lower precision compared to 32-bit float can result in two times reduced memory consumption. You can either try setting `--precision 16-true` for regular 16-bit precision or  `--precision bf16-true` if your GPU supports brain-float 16-bit precision. ([This brief video](https://lightning.ai/courses/deep-learning-fundamentals/9.0-overview-techniques-for-speeding-up-model-training/unit-9.1-accelerated-model-training-via-mixed-precision-training/) explains the difference between regular 16-bit and bf16-bit precision.) 
+
```
Comment (Andrei-Aksionov) [inline tutorials/quantize.md:24]: A small nit-picking: avoid multiple empty lines where only one should be used.
https://github.com/DavidAnson/markdownlint/blob/v0.29.0/doc/md012.md

File: tutorials/quantize.md [hunk 5]
```diff
@@ -1,14 +1,32 @@
...
-However, this might not be enough for large models or when using GPUs with limited memory.
+This document provides different strategies for quantizing the various models available in Lit-GPT to reduce GPU memory usage, which is useful for running larger models on certain GPU hardware. 
+
+**All the examples below were run on an A100 40GB GPU.**
 
 > **Note**:
 > Quantization is only supported with inference (generate and chat scripts).
 
+
 ### Baseline
```
Comment (Andrei-Aksionov) [inline tutorials/quantize.md:11]: I understand that it was like that before, but other "chapters" uses only double # sign.

**Comment by Andrei-Aksionov** [comment]
Answer to the question no one asked: I use [Markdownlint](https://marketplace.visualstudio.com/items?itemName=DavidAnson.vscode-markdownlint) extension for VSCode to highlight and auto-fix linting errors.

File: tutorials/quantize.md (hunk 3 above)
Comment (rasbt) [reply tutorials/quantize.md:21]: Thanks, somehow my local markdown editor didn't complain about this but this is of course an issue! Submitted a PR to fix this and the other cosmetics you highlighted.

**Pull request merged**